quickwit/quickwit-metastore/src/split_metadata.rs

```diff
@@ -243,6 +263,42 @@ impl FromStr for SplitState {
     }
 }
 
+/// `SplitMaturity` defines the maturity of a split, it is either `Mature`
+/// or `Immature` with a given maturation period.
+/// The maturity is determined by the `MergePolicy`.
+#[derive(Clone, Copy, Debug, Default, Eq, Serialize, Deserialize, PartialEq)]
+#[serde(tag = "type")]
+#[serde(rename_all = "snake_case")]
+pub enum SplitMaturity {
+    /// The split is mature and no longer a candidates for merges.
+    #[default]
+    Mature,
+    /// The split is immature and can undergo merges until `ripening_period` passes,
+    /// measured relatively from the split's creation timestamp.
+    Immature {
+        /// Time to maturity.
+        #[serde(with = "serde_duration_millisecs")]
+        #[serde(rename = "maturation_period_millisecs")]
+        maturation_period: Duration,
+    },
+}
+
+// Serializer/Deserializer of `Duration` in milliseconds.
+pub mod serde_duration_millisecs {
```

Comment:
serde_with has an implem too if you want to add the dependency (I agree it is always a tiny bit sad)

I'd prefer secs than millisecs here. I'm not too fond of reading pinball high scores. :)

Reply:
Actually, there is a plan led by @guilload about putting all split metadata "timestamps"  in milliseconds and adding a small wrapper `Timestamp` to avoid having `i64` everywhere.

So we will eventually move to a world where everything is in milliseconds. Thus the duration in millis.

Reply:
Milliseconds are much better for unit testing. We have a bunch of time::sleep(1s) in our metastore unit tests, making them slow and flaky, because of the poor resolution of our timestamps.

Reply:
For this particular duration, it does not impact the unit tests.

On the contrary, we don't control `create_timestamp`, `update_timestamp`, and `publish_timestamp` values, thus the need for `sleep`.